ios/RNCWKWebView.m

```diff
@@ -127,6 +127,68 @@ - (void)didMoveToWindow
+        if(!_incognito && !_cacheEnabled) {
+          wkWebViewConfig.websiteDataStore = [WKWebsiteDataStore nonPersistentDataStore];
+        }
+        for (NSHTTPCookie *cookie in [[NSHTTPCookieStorage sharedHTTPCookieStorage] cookies]) {
+          [wkWebViewConfig.websiteDataStore.httpCookieStore setCookie:cookie completionHandler:nil];
+        }
+      } else {
+        NSMutableString *script = [NSMutableString string];
+
+        // Clear all existing cookies in a direct called function. This ensures that no
+        // javascript error will break the web content javascript.
+        // We keep this code here, if someone requires that Cookies are also removed within the
+        // the WebView and want to extends the current sharedCookiesEnabled option with an
+        // additional property.
+        // Generates JS: document.cookie = "key=; Expires=Thu, 01 Jan 1970 00:00:01 GMT;"
+        // for each cookie which is already available in the WebView context.
+        /*
+        [script appendString:@"(function () {\n"];
+        [script appendString:@"  var cookies = document.cookie.split('; ');\n"];
+        [script appendString:@"  for (var i = 0; i < cookies.length; i++) {\n"];
+        [script appendString:@"    if (cookies[i].indexOf('=') !== -1) {\n"];
+        [script appendString:@"      document.cookie = cookies[i].split('=')[0] + '=; Expires=Thu, 01 Jan 1970 00:00:01 GMT';\n"];
+        [script appendString:@"    }\n"];
+        [script appendString:@"  }\n"];
+        [script appendString:@"})();\n\n"];
+        */
+
+        // Set cookies in a direct called function. This ensures that no
+        // javascript error will break the web content javascript.
+          // Generates JS: document.cookie = "key=value; Path=/; Expires=Thu, 01 Jan 20xx 00:00:01 GMT;"
+        // for each cookie which is available in the application context.
+        [script appendString:@"(function () {\n"];
+        for (NSHTTPCookie *cookie in [[NSHTTPCookieStorage sharedHTTPCookieStorage] cookies]) {
+          [script appendFormat:@"document.cookie = %@ + '=' + %@",
+            RCTJSONStringify(cookie.name, NULL),
+            RCTJSONStringify(cookie.value, NULL)];
+          if (cookie.path) {
+            [script appendFormat:@" + '; Path=' + %@", RCTJSONStringify(cookie.path, NULL)];
+          }
+          if (cookie.expiresDate) {
+            [script appendFormat:@" + '; Expires=' + new Date(%f).toUTCString()",
+              cookie.expiresDate.timeIntervalSince1970 * 1000
+            ];
+          }
+          [script appendString:@";\n"];
+        }
+        [script appendString:@"})();\n"];
+
+        WKUserScript* cookieInScript = [[WKUserScript alloc] initWithSource:script
+                                                              injectionTime:WKUserScriptInjectionTimeAtDocumentStart
+                                                           forMainFrameOnly:NO];
```

Comment:
forMainFrameOnly:NO

Isn't this a risk to leak your cookies to anything loaded by your website? I'm not sure 🤔

Reply:
Yep indeed, 
From Docs: `A Boolean value indicating whether the script should be injected only into the main frame (true) or into all frames (false).`
It should not be necessary to provide the cookie to all frames. 
@jerolimov could you give this a try? I could not test on devices with iOS < 11 atm. Thank your very much.

Reply:
@Titozzz I changed the code to `forMainFrameOnly: YES` seems the logic way. 
I tested on an old iPad with iOS 9.x, seemed to work as expected.